solar_system.rb

```diff
@@ -0,0 +1,352 @@
+    planet_num = 0
+    @planets.each do |planet|
+      planet_num += 1
+      planet_name << "#{planet_num}. #{planet.name} \n"
+    end
+    return planet_name
+  end
+
+  def planet_facts
+    puts "Here are some facts about planets:"
+    @planets.each do |planet|
+      puts "#{@planets}"
+    end
+  end
+
+end
+
+planets = []
+
+jupiter = {:name => "Jupiter", :moons => 67, :surface_temp_c => -148, :diameter => "142984 km", :orbit_period => "4333 days", :length_of_day => "10 hours " }
+mars = {:name => "Mars", :moons => 2, :surface_temp_c => "-153 to 20" , :diameter => "6792 km", :orbit_period => "687 days", :length_of_day => " 25 hours" }
+venus= {:name => "Venus", :moons => 0, :surface_temp_c => 426, :diameter => "12104 km", :orbit_period => "225 days", :length_of_day => "117 earth days"}
+
+planets << Planet.new(jupiter)
+planets << Planet.new(mars)
+planets << Planet.new(venus)
+
+outer_space = SolarSystem.new(planets)
+
+puts "Welcome the Solar System app!"
+puts outer_space.planet_list
+
+interactions = ["EXPLORE","NEW","EXIT"]
+
+run_once = false
+interact = ""
+
+until interact == "EXIT"
+
+  # IF this is the first time the user is prompted to enter an action
+  # OR they chose a valid options THEN display the regular prompts
+  if (run_once == false || interactions.include?(interact))
+
+    if (run_once == false)
+      puts "To learn more about a specific planet type: Explore"
+      puts "Discovered a new planet? Type: New"
+      puts "Had enough? To quit type: Exit"
+    else
+      puts "What would you like to do (new, explore, exit)?"
+    end
+
+  # ELSE the loop has already run and the user supplied invalid input,
+  # so we need to prompt them to enter something valid.
+  else
+
+    puts "Please enter a valid option."
+
+  end
+
+  interact = gets.chomp.upcase
+
+  run_once = true
+
+  case interact
+    when "EXPLORE"
+      puts outer_space.planet_list
+      puts "Type the number that corresponds to the planet you would like to learn more about:"
+
+      # TO DO: Add a check to ensure they've supplied a valid number
+
+      puts planets[gets.chomp.to_i - 1].all_facts
+
+    when "NEW"
+      new_user_planet = {}
+      print "What is the planet name? "
+      new_user_planet[:name] = gets.chomp.to_s.capitalize
+      print "How many moons does the planet have? "
+      new_user_planet[:moons] = gets.chomp.to_i
+      print "What is the planet's suraface temperature in celcius? "
+      new_user_planet[:surface_temp_c] = gets.chomp.to_s
+      print "What is the planet's diameter in kilometers? "
+      new_user_planet[:diameter] = gets.chomp.to_i
+      print "What is the planet's orbit period? "
+      new_user_planet[:planet_orbit] = gets.chomp.to_s
+      print "How long is the solar day? "
+      new_user_planet[:length_of_day] = gets.chomp.to_s
+      print "How far away from the sun is the planet? "
+      new_user_planet[:distance_from_sun] = gets.chomp.to_s
+
+      planets << Planet.new(new_user_planet)
```

Comment:
I mentioned this in my review but I think it'd be great if SolarSystem had this method:
```ruby
class SolarSystem
...
def add_planet( new_planet )
  @planets << new_planet
end
```
and this line was instead
```ruby
outer_space.add_planet( Planet.new(new_user_planet) )
```